// @aws-amplify/ui/dist/types/theme/tokens/components/badge.d 2.ts
import { DesignTokenProperties, OutputVariantKey } from '../types/designToken';
declare type VariationKey = 'backgroundColor' | 'color';
declare type BadgeVariationTokens<OutputType> = DesignTokenProperties<VariationKey, OutputType>;
declare type SizeKey = 'fontSize' | 'paddingHorizontal' | 'paddingVertical';
declare type BadgeSizeTokens<OutputType> = DesignTokenProperties<SizeKey, OutputType>;
declare type BadgeKey = 'backgroundColor' | 'borderRadius' | 'color' | 'fontSize' | 'fontWeight' | 'lineHeight' | 'paddingHorizontal' | 'paddingVertical' | 'textAlign';
export declare type BadgeTokens<OutputType extends OutputVariantKey> = DesignTokenProperties<BadgeKey, OutputType> & {
    error?: BadgeVariationTokens<OutputType>;
    large?: BadgeSizeTokens<OutputType>;
    small?: BadgeSizeTokens<OutputType>;
    info?: BadgeVariationTokens<OutputType>;
    success?: BadgeVariationTokens<OutputType>;
    warning?: BadgeVariationTokens<OutputType>;
};
export declare const badge: Required<BadgeTokens<'default'>>;
export {};
