// @aws-amplify/ui/dist/types/theme/tokens/components/table.d 2.ts
import { DesignTokenProperties, OutputVariantKey } from '../types/designToken';
declare type BaseTableTokens<Output> = DesignTokenProperties<'display' | 'verticalAlign', Output>;
declare type TableCellTokens<Output> = DesignTokenProperties<'borderColor' | 'borderStyle' | 'borderWidth' | 'color' | 'display' | 'fontSize' | 'fontWeight' | 'padding' | 'verticalAlign', Output> & {
    large?: TableCellSizeTokens<Output>;
    small?: TableCellSizeTokens<Output>;
};
declare type TableCellSizeTokens<Output> = DesignTokenProperties<'fontSize' | 'padding', Output>;
declare type TableCaptionSizeTokens<Output> = DesignTokenProperties<'fontSize', Output>;
export declare type TableTokens<Output extends OutputVariantKey> = DesignTokenProperties<'borderCollapse' | 'display' | 'width', Output> & {
    head?: BaseTableTokens<Output>;
    body?: BaseTableTokens<Output>;
    foot?: BaseTableTokens<Output>;
    row?: BaseTableTokens<Output> & {
        hover?: DesignTokenProperties<'backgroundColor', Output>;
        striped?: DesignTokenProperties<'backgroundColor', Output>;
    };
    header?: TableCellTokens<Output>;
    data?: TableCellTokens<Output>;
    caption?: DesignTokenProperties<'captionSide' | 'color' | 'display' | 'fontSize' | 'textAlign' | 'wordBreak', Output> & {
        large?: TableCaptionSizeTokens<Output>;
        small?: TableCaptionSizeTokens<Output>;
    };
};
export declare const table: Required<TableTokens<'default'>>;
export {};
