// @aws-amplify/ui/dist/types/helpers/storage/fileUploader/utils/uploader.d 2.ts
import { StorageAccessLevel } from '@aws-amplify/storage';
export declare function uploadFile({ file, fileName, level, progressCallback, errorCallback, completeCallback, isResumable, ...rest }: {
    file: File;
    fileName: string;
    level: StorageAccessLevel;
    isResumable?: boolean;
    progressCallback: (progress: {
        loaded: number;
        total: number;
    }) => void;
    errorCallback: (err: string) => void;
    completeCallback: (event: any) => void;
}): Promise<void> | import("@aws-amplify/storage").UploadTask;
/**
 * Format bytes as human-readable text.
 *
 * @param bytes Number of bytes.
 * @param si True to use metric (SI) units, aka powers of 1000. False to use
 *           binary (IEC), aka powers of 1024.
 * @param dp Number of decimal places to display.
 *
 * @return Formatted string.
 */
export declare function humanFileSize(bytes: any, si?: boolean, dp?: number): string;
export declare const checkMaxSize: (maxSize: number, file: File) => string | null;
export declare const returnAcceptedFiles: (files: File[], acceptedFileTypes: string[]) => File[];
export declare const isValidExtension: (fileName: string, fileName2: string) => boolean;
