// @aws-amplify/ui-angular/esm2015/lib/primitives/button/button.component.js
import { Component, HostBinding, Input } from '@angular/core';
import * as i0 from "@angular/core";
export class ButtonComponent {
    constructor() {
        this.type = 'button';
        this.fullWidth = false;
        this.isDisabled = false;
        this.size = 'medium';
        this.variation = 'default';
        this.fontWeight = 'normal';
    }
    get getDisabled() {
        return this.isDisabled ? '' : null;
    }
    get classNames() {
        let className = 'amplify-button';
        if (this.variation) {
            className += ` amplify-button--${this.variation}`;
        }
        if (this.size) {
            className += ` amplify-button--${this.size}`;
        }
        if (this.fullWidth) {
            className += ` amplify-button--fullwidth`;
        }
        if (this.isDisabled) {
            className += ` amplify-button--disabled amplify-button--loading`;
        }
        return className;
    }
    ngOnInit() {
        this.typeAttr = this.type;
        this.fullWidthAttr = this.fullWidth;
        this.sizeAttr = this.size;
        this.variationAttr = this.variation;
        this.fontWeightAttr = this.fontWeight;
    }
}
ButtonComponent.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ButtonComponent, deps: [], target: i0.ɵɵFactoryTarget.Component });
ButtonComponent.ɵcmp = i0.ɵɵngDeclareComponent({ minVersion: "12.0.0", version: "12.2.16", type: ButtonComponent, selector: "button[amplify-button]", inputs: { type: "type", fullWidth: "fullWidth", isDisabled: "isDisabled", size: "size", variation: "variation", fontWeight: "fontWeight" }, host: { properties: { "type": "this.typeAttr", "attr.data-fullwidth": "this.fullWidthAttr", "attr.data-size": "this.sizeAttr", "attr.data-variation": "this.variationAttr", "style.font-weight": "this.fontWeightAttr", "attr.disabled": "this.getDisabled", "class": "this.classNames" } }, ngImport: i0, template: "<ng-content></ng-content>\n" });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "12.2.16", ngImport: i0, type: ButtonComponent, decorators: [{
            type: Component,
            args: [{
                    selector: 'button[amplify-button]',
                    templateUrl: './button.component.html',
                }]
        }], propDecorators: { type: [{
                type: Input
            }], fullWidth: [{
                type: Input
            }], isDisabled: [{
                type: Input
            }], size: [{
                type: Input
            }], variation: [{
                type: Input
            }], fontWeight: [{
                type: Input
            }], typeAttr: [{
                type: HostBinding,
                args: ['type']
            }], fullWidthAttr: [{
                type: HostBinding,
                args: ['attr.data-fullwidth']
            }], sizeAttr: [{
                type: HostBinding,
                args: ['attr.data-size']
            }], variationAttr: [{
                type: HostBinding,
                args: ['attr.data-variation']
            }], fontWeightAttr: [{
                type: HostBinding,
                args: ['style.font-weight']
            }], getDisabled: [{
                type: HostBinding,
                args: ['attr.disabled']
            }], classNames: [{
                type: HostBinding,
                args: ['class']
            }] } });
